### PR TITLE
Extracts adding PURL and assigning file/fileset identifiers to CocinaObjectStore

### PR DESCRIPTION
**Pull request by justinlittman**
## Why was this change made? 🤔
These updates need to happen regardless of datastore.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

Unit

### CODE REVIEW

File: app/services/cocina/object_creator.rb [hunk 1]
```diff
@@ -143,8 +143,7 @@ def catkey_for(obj)
     # @raises SymphonyReader::ResponseError if symphony connection failed
     def add_description(fedora_object, cocina_object, trial:)
       if cocina_object.description
-        description = AddPurlToDescription.call(cocina_object.description, fedora_object.pid)
-        fedora_object.descMetadata.content = Cocina::ToFedora::Descriptive.transform(description, fedora_object.pid).to_xml
+        fedora_object.descMetadata.content = Cocina::ToFedora::Descriptive.transform(cocina_object.description, fedora_object.pid).to_xml
```
Comment (ndushay) [inline app/services/cocina/object_creator.rb:146]: "pid"

File: spec/services/cocina_object_store_spec.rb [hunk 2]
```diff
@@ -9,6 +9,7 @@
     let(:item) { instance_double(Dor::Item) }
     let(:date) { Time.zone.now }
     let(:cocina_object) { instance_double(Cocina::Models::DRO, externalIdentifier: druid) }
+
```
Comment (ndushay) [inline spec/services/cocina_object_store_spec.rb:12]: whoops (extra line)

**Review by ndushay — APPROVED** [review]
Not sure I fully grok it;  approving;  3 comments for your consideration.

File: app/services/cocina/object_creator.rb (hunk 1 above)
Comment (jcoyne) [reply app/services/cocina/object_creator.rb:146]: Can't really do anything about the fact that he's calling ActiveFedora::Base#pid

File: spec/services/cocina/object_creator_spec.rb [hunk 3]
```diff
@@ -78,10 +78,6 @@
                 type: 'email',
                 value: 'marypoppins@umbrellas.org',
                 displayLabel: 'Contact'
-              },
-              {
-                type: 'preferred citation',
-                value: 'Zappa, F. (2013) :link:'
```
[the remark this answers is not in the document]
Comment (justinlittman) [reply spec/services/cocina/object_creator_spec.rb:84]: Yes, it is moved to a different spec.

File: spec/services/cocina_object_store_spec.rb (hunk 2 above)
Comment (justinlittman) [reply spec/services/cocina_object_store_spec.rb:12]: Removed.

**Pull request merged**